core-kotlin-modules/core-kotlin-advanced-2/src/test/kotlin/callback/CallbackFunctionTest.kt

```diff
@@ -0,0 +1,103 @@
+package callback
+
+import org.junit.jupiter.api.Test
+import kotlin.test.assertEquals
+
+data class User(var firstName: String, var lastName: String)
+class CallbackFunctionTest {
+
+    @Test
+    fun `callback function to perform LCM`(){
+        val lcm = {x: Int, y: Int ->
+            var gcd = 1
+
+            var i = 1
+            while (i <= x && i <= y) {
+                // Checks if i is factor of both integers
+                if (x % i == 0 && y % i == 0)
+                    gcd = i
+                ++i
+            }
+
+            x * y / gcd
+
+        }
+
+        fun getLCM(x: Int, y: Int, lcm: (Int, Int) -> Int): Int {
+            return lcm(x, y)
+        }
+
+        var res1 = getLCM(72, 120, lcm)
+        var res2 = getLCM(2, 3, lcm)
+        var res3 = getLCM(47, 28, lcm)
+
+        assertEquals(360, res1)
+        assertEquals(6, res2)
+        assertEquals(1316, res3)
+    }
+
+    @Test
+    fun `asynchronous callback to load remote data`(){
+
+        fun loadUsersFromServer(callback: (List<User>) -> Unit) {
+            Thread.sleep(5000)
+            val users = listOf(User("Flore", "P"), User("Nappy", "Sean"), User("Ndole", "Paul"))
+            callback(users)
+        }
+
+        var listofUsers = emptyList<User>()
+        fun executeLoading(){
+            loadUsersFromServer { users ->
+                listofUsers = users
+            }
+
+        }
+
+        executeLoading()
+        assertEquals(3, listofUsers.size)
+        assertEquals("Flore", listofUsers[0].firstName)
+        assertEquals("Sean", listofUsers[1].lastName)
+        assertEquals("Ndole Paul", "${listofUsers[2].firstName} ${listofUsers[2].lastName}" )
+
+    }
+
+    @Test
+    fun `demonstration of callback hell`(){
```

Comment:
It may make more sense to show this example in it's own file, outside of a unit test?